addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -3931,6 +3931,92 @@ options.registry.GalleryElement = options.Class.extend({
     },
 });
 
+options.registry.Button = options.Class.extend({
+    forceDuplicateButton: true,
+
+    /**
+     * @override
+     */
+    onBuilt(options) {
+        // Only if the button is built, not if a snippet containing that button
+        // is built (e.g. true if dropping a button from the snippet menu onto
+        // the page, false if dropping an "image-text" snippet).
+        if (options.isCurrent) {
+            this._adaptButtons();
+        }
+    },
+    /**
+     * @override
+     */
+    onClone(options) {
+        // Only if the button is cloned, not if a snippet containing that button
+        // is cloned.
+        if (options.isCurrent) {
+            this._adaptButtons();
+        }
+    },
```

Comment:
This is buggy to me: if you have a primary button followed by a secondary one, then you duplicated the primary one... it becomes a secondary one, it does not make sense to be. When duplicating I think you only want the part about adding the spaces, no? Not changing the button style?

To be changed after the merge.